packages/@lwc/compiler/src/transformers/template.ts

```diff
@@ -80,7 +80,9 @@ function serialize(
     buffer += 'if (_implicitScopedStylesheets) {\n';
     buffer += `  tmpl.stylesheets.push.apply(tmpl.stylesheets, _implicitScopedStylesheets)\n`;
     buffer += `}\n`;
-    buffer += `tmpl.stylesheetToken = "${scopeToken}"\n`;
+    buffer += 'if (_implicitStylesheets || _implicitScopedStylesheets) {\n';
+    buffer += `  tmpl.stylesheetToken = "${scopeToken}"\n`;
+    buffer += '}\n';
```

Comment:
An additional benefit for no-stylesheet components is that the above code gets tree-shaken out by Rollup/Terser. The end result is just:

```js
tmpl.stylesheets = [];
```

Unfortunately neither Rollup nor Terser is smart enough to apply the _opposite_ optimization for components that have stylesheets:

```js
// input
tmpl.stylesheets = [stylesheet];
if (_implicitScopedStylesheets) {
  tmpl.stylesheets.push.apply(tmpl.stylesheets, _implicitScopedStylesheets);
}
if (_implicitScopedStylesheets) {
  tmpl.stylesheetToken = "x-app_app";
}
```

```js
// optimal output
tmpl.stylesheets = [stylesheet];
tmpl.stylesheets.push.apply(tmpl.stylesheets, _implicitScopedStylesheets);
tmpl.stylesheetToken = "x-app_app";
```

(The actual output retains the `if()` statements.)

I consider this unfortunate but not a big deal.

Reply:
This is neat, I would also maybe do something like:

```js
if (_implicitStylesheets || _implicitScopedStylesheets) {
  tmpl.stylesheet = createStyleSheet({
    token: "x-app_app"
    sheets: [_implicitStylesheets, _implicitScopedStylesheets]
  })
}
```

For no-stylesheet templates, this would allow the whole thing to be tree-shaken, which is kind of nice.